object.go

```diff
@@ -757,6 +757,35 @@ func (o *Object) ValueNotEqual(key string, value interface{}) *Object {
 	return o.NotValueEqual(key, value)
 }
 
+// Iter returns a new map of Value
+//
+// Example:
+//
+// object := newObject(chain, map[string]interface{})
+//
+//	for n, val := range object.Iter() {
+//		    val.Equal(object[n])
+//		}
+func (o *Object) Iter() map[string]Value {
+	o.chain.enter("Iter")
+	defer o.chain.leave()
+
+	if o.chain.failed() {
+		return map[string]Value{}
+	}
+
+	obj := make(map[string]Value)
+
+	for k, v := range o.value {
+		valueChain := o.chain.clone()
+		valueChain.enter("Iter[%d]", v)
```

Comment:
* Let's change `enter` to `replace`; this will substitute last part of path from `Iter()` to `Iter[...]`
* Instead of `v`, we should use `k` (our assertion paths include keys, not values)
* Instead of `%d`, we should use `%q` (we use this format for keys)